lib/vnet/daemon/client_darwin.m

```diff
@@ -35,38 +35,37 @@
 //
 // The filename and the value of the Label key in the plist file and the Mach
 // service of of the daemon must match the string returned from this function.
-NSString *DaemonLabel(void) {
-  NSBundle *main = [NSBundle mainBundle];
+NSString *DaemonLabel(NSString *bundlePath) {
+  NSBundle *main = [NSBundle bundleWithPath:bundlePath];
   if (!main) {
     return @"";
   }
 
   NSString *bundleIdentifier = [main bundleIdentifier];
-  if ([bundleIdentifier length] == 0) {
-    return bundleIdentifier;
+  if (!bundleIdentifier || [bundleIdentifier length] == 0) {
+    return @"";
   }
 
   return [NSString stringWithFormat:@"%@.vnetd", bundleIdentifier];
 }
 
 // DaemonPlist takes the result of DaemonLabel and appends ".plist" to it
 // if not empty.
-NSString *DaemonPlist(void) {
-  NSString *label = DaemonLabel();
+NSString *DaemonPlist(NSString *bundlePath) {
+  NSString *label = DaemonLabel(bundlePath);
   if ([label length] == 0) {
     return label;
   }
 
   return [NSString stringWithFormat:@"%@.plist", label];
 }
 
-// TODO: This needs to accept bundle_path as an argument in order to with symlinks.
-void RegisterDaemon(RegisterDaemonResult *outResult) {
+void RegisterDaemon(const char *bundle_path, RegisterDaemonResult *outResult) {
   if (@available(macOS 13, *)) {
     SMAppService *service;
     NSError *error;
 
-    service = [SMAppService daemonServiceWithPlistName:(DaemonPlist())];
+    service = [SMAppService daemonServiceWithPlistName:(DaemonPlist(@(bundle_path)))];
```

Comment:
Do the @(bundle_path) conversion assume UTF-8?

Reply:
`@(bundle_path)` does assume UTF-8. Do I need to do anything extra because of that?

https://clang.llvm.org/docs/ObjectiveCLiterals.html#boxed-expressions
https://developer.apple.com/documentation/foundation/nsstring/1497379-stringwithutf8string

Reply:
Nope, just checking. Thanks for the links.